app/models/user.rb

```diff
@@ -525,6 +513,21 @@ def hotness_score
     search_score
   end
 
+  def connected_authentication_providers
+    identities_enabled.pluck(:provider).map(&:to_sym)
+  end
+
+  def authenticated_social_accounts
+    user_identities = identities_enabled
+    return {} if user_identities.blank?
+
+    urls = user_identities.pluck(:auth_data_dump)
+      .map { |data| data.dig(:info, :urls) }
```

Comment:
In Ruby when you see a `map` plus `reduce` combination like this you can often use `each_with_object`. Something like this completely untested code:

```ruby
user_identities.pluck(:auth_data_dump).each_with_object({}) do |data, hash|
  hash.merge(data.dig(:info, :urls))
end